strongbox-commons/src/test/java/org/carlspring/strongbox/booters/SystemPropertiesInitializerTest.java

```diff
@@ -0,0 +1,174 @@
+package org.carlspring.strongbox.booters;
+
+import java.util.Properties;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Configuration;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.carlspring.strongbox.booters.SystemPropertiesInitializer.STRONGBOX_HOME_PROPERTY_NAME;
+import static org.carlspring.strongbox.booters.SystemPropertiesInitializer.STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME;
+import static org.carlspring.strongbox.booters.SystemPropertiesInitializer.STRONGBOX_VAULT_PROPERTY_NAME;
+
+class SystemPropertiesInitializerTest
+{
+    private static final String HOME_DIR_VALUE = "home5150";
+    private static final String HOME_DIR_PROPERTY = STRONGBOX_HOME_PROPERTY_NAME + "=" + HOME_DIR_VALUE;
+    private static final String VAULT_DIR_VALUE = "vault5150";
+    private static final String VAULT_DIR_PROPERTY = STRONGBOX_VAULT_PROPERTY_NAME + "=" + VAULT_DIR_VALUE;
+    private static final String STORAGE_BASEDIR_VALUE = "storage5150";
+    private static final String STORAGE_BASEDIR_PROPERTY = STRONGBOX_STORAGE_BASEDIR_PROPERTY_NAME + "=" + STORAGE_BASEDIR_VALUE;
+
+    private final ApplicationContextRunner contextRunner = new ApplicationContextRunner();
+    private final SystemPropertiesInitializer initializer = new SystemPropertiesInitializer();
+    private Properties originalProps;
+
+    @BeforeEach
+    void rememberAndClearOutSystemProperties()
+    {
+        originalProps = System.getProperties();
+        clearRelevantSystemProperties();
+    }
+
+    @AfterEach
+    void replaceSystemProperties()
+    {
+        System.setProperties(originalProps);
+    }
+
+    @Test
+    void testSystemPropertiesAreInitialized()
+    {
+        contextRunner.withUserConfiguration(EmptyConfiguration.class)
```

Comment:
If you have not seen `ApplicationContextRunner` yet - you can thank me later. This thing has turned out to be super helpful for testing things like this. Also note that its a super thin slice of the context, runs way way faster than `@SpringBootTest`.